controllers/operatorpolicy_controller.go

```diff
@@ -121,16 +125,84 @@ func (r *OperatorPolicyReconciler) PeriodicallyExecOperatorPolicies(freq uint, e
 func (r *OperatorPolicyReconciler) handleSinglePolicy(
 	policy *policyv1beta1.OperatorPolicy,
 ) error {
-	policy.Status.ComplianceState = policyv1.Compliant
+	OpLog.Info("Logging operator policy", "policy", policy.Name)
 
-	err := r.updatePolicyStatus(policy)
-	if err != nil {
-		OpLog.Info("error while updating policy status")
+	subscriptionSpec := r.buildSubscription(policy)
+	err := r.Get(context.TODO(),
+		types.NamespacedName{Namespace: subscriptionSpec.Namespace, Name: subscriptionSpec.Name},
+		subscriptionSpec)
+	exists := err == nil
+	shouldExist := strings.EqualFold(string(policy.Spec.ComplianceType), string(policyv1.MustHave))
 
-		return err
+	// Object does not exist but it should exist, create object
+	if !exists && shouldExist {
+		if strings.EqualFold(string(policy.Spec.RemediationAction), string(policyv1.Enforce)) {
+			OpLog.Info("creating kind " + subscriptionSpec.Kind + " in ns " + subscriptionSpec.Namespace)
+			err = r.Create(context.TODO(), subscriptionSpec)
+
+			if err != nil {
+				r.setCompliance(policy, policyv1.NonCompliant)
+				OpLog.Error(err, "Could not handle missing musthave object")
+
+				return err
+			}
+
+			operatorGroup := r.buildOperatorGroup(policy)
+			err = r.Create(context.TODO(), operatorGroup)
```

Comment:
This is fine for the initial prototype, but eventually it will need to consider what to do when there is already an operator group in the namespace. Just a comment here so we don't forget 😀